.github/workflows/posix.yml

```diff
@@ -0,0 +1,14 @@
+name: POSIX Builds
```

Comment:
So, the structure will be: one .yml per platform?
Other option might be, one .yml per build config, or just one .yml for all builds.

Also IDK: is there 1-to-1 matching between .yml and workflow? Are these workflows parallelised?

Reply:
Also: these question does not need to be answered to submit this change since the structure can be changed later. I just ask to understand GHA.

Reply:
I wasn't quite sure how it will pan out in the long run.  I was just going to start with POSIX and then either continue with other target platforms in the same workflow (and rename the workflow) or create different ones for FREERTOS, etc.

There is a 1:1 matching between yml and workflows.  Workflows can trigger other workflows, though, and runners are assigned to any waiting worfklow which means workflows can run in parallel. Additionally each workflow can have a job matrix inside of it with each job running in parallel.   

My plan is to write up a doc describing all of this once I get at least one build going.

Reply:
Perfect, thx!